test/integration/server-discovery-and-monitoring/server_discovery_and_monitoring_logging.test.ts

```diff
@@ -0,0 +1,69 @@
+import { expect } from 'chai';
+import { setTimeout } from 'timers/promises';
+
+import { Binary, type MongoDBLogWritable, type ServerHeartbeatSucceededEvent } from '../../mongodb';
+
+describe('SDAM Logging Integration Tests', function () {
+  const monitoringEvents: Array<ServerHeartbeatSucceededEvent> = [];
+  const loggingEvents: Array<{ reply: string }> = [];
+
+  beforeEach(async function () {
+    const logger: MongoDBLogWritable = {
+      write(log) {
+        if (log.message === 'Server heartbeat succeeded') {
+          loggingEvents.push(log as unknown as { reply: string });
+        }
+      }
+    };
+    const client = this.configuration.newClient(
+      {},
+      {
+        mongodbLogPath: logger,
+        mongodbLogComponentSeverities: { topology: 'trace' },
+        [Symbol.for('@@mdb.enableMongoLogger')]: true
+      }
+    );
+
+    client.on('serverHeartbeatSucceeded', monitoringEvents.push.bind(monitoringEvents));
+
+    await client.connect();
+    // give the driver a chance to connect to all servers and collect some heartbeats
+    await setTimeout(100);
```

Comment:
LGTM, but just a knowledge question, why is 100ms a sufficient time?

Reply:
Also because you might notice this sleep is after the awaited `connect`.  Connect will (by default) resolve when the driver finds a primary.  That means that `connect` can resolve before connecting to secondaries, arbiters or other servers.